hooks/command

```diff
@@ -158,6 +158,20 @@ while IFS='=' read -r name _ ; do
   fi
 done < <(env | sort)
 
+# Propagate all environment variables into the container if requested
+if [[ "${BUILDKITE_PLUGIN_DOCKER_PROPAGATE_ENVIRONMENT:-false}" =~ ^(true|on|1)$ ]] ; then
+  if [[ -n "${BUILDKITE_ENV_FILE:-}" ]] ; then
+    # Read in the env file and convert to --env params for docker
+    # This is because --env-file doesn't support newlines or quotes per https://docs.docker.com/compose/env-file/#syntax-rules
+    while read -r var; do
```

Comment:
My last concern is that this will load in the value of env as it is at run time, vs what it was when things started. This might be confusing. 

For instance, if the ENV_FILE has llamas=always and a pre-command hook sets it to llamas=sometimes, I'd expect the initial value. 

Thoughts?

Reply:
I'd expect the updated value, would make debugging a lot easier